### ISSUE TITLE
Suppress partially unknown type errors

### ISSUE THREAD

**Issue by multimeric** [microsoft/pyright#8154]
**Is your feature request related to a problem? Please describe.**

Sometimes I want to enforce type checking, but to hold my code to a somewhat lower standard by allowing partially unknown types.

Currently, if you have a function like the following:
```python
def foo(arg: dict):
    pass
```

Pyright reports:
```
$ pyright .
foo.py
  foo.py:1:9 - error: Type of parameter "arg" is partially unknown
    Parameter type is "dict[Unknown, Unknown]" (reportUnknownParameterType)
  foo.py:1:14 - error: Expected type arguments for generic class "dict" (reportMissingTypeArgument)
2 errors, 0 warnings, 0 informations
```

However, if I set `reportUnknownParameterType = false`, then I'm also allowing functions like this, which are completely uninformative:
```python
def foo(arg):
    pass
```

**Describe the solution you’d like**

I think it would be great if a separate config option were provided for this group of errors, that would suppress partially unknown errors but still raise fully unknown errors:

```
reportPartiallyMissingTypeArgument = false
reportPartiallyUnknownParameterType = false
```

**Comment by erictraut** [comment]
We have on a couple of occasions in the past split existing checks into two or more finer-grained checks, but the bar is quite high. This is disruptive to existing pyright users because it's effectively a "breaking change" if you have already configured your project.

The proposal would also require splitting all of the `reportUnknown...` checks for consistency, which would result in a bunch of new checks.

For these reasons, I'm going to reject this feature request for now. I'm willing to reconsider in the future if there's signal (e.g. thumbs up reactions to this issue) from many other pyright users that it's something they want to see.

**Comment by multimeric** [comment]
Understood.

Incidentally, would it be possible to treat `reportMissingTypeArgument` as a category with identical behaviour, and then divide it into two subsets: `reportPartiallyMissingTypeArgument`and `reportFullyMissingTypeArgument` that can also be configured separately? This would prevent it being a breaking change.

**Comment by dibrinsofor** [comment]
It sounds like they want tag checks; maybe having a flag for this specific feature won't break backwards compatibility. I've noticed this shows up with decorators and higher-order functions that do not exactly care about the function's signature.

```python
def _count_parameters(func: Callable) -> int:
  return len(signature(func).parameters)
```
